### DISCUSSION TITLE
iRobot Create 3 is stuck in Boot loop

### DISCUSSION THREAD

**Discussion by Affan60** [iRobotEducation/create3_docs#61] [Questions] [answered]
The iRobot create 3 is stuck in booting loop. The white ring keeps swirling and it does not complete the booting process. If I press the center button, it does not power off or do anything. is there any solution for that?

**Comment by justinIRBT** [comment]
@Affan60 To confirm, your robot never plays any music (it has a multi note jingle it plays when the boot process finishes.).  Did your robot ever boot successfully, were you able to go through the setup process at some point where you updated your firmware and connected your robot to the network as described here:
https://edu.irobot.com/create3-setup

If your robot is on the network, can you try pulling the logs from the robot to help us diagnose.  First, you will need to access the robot’s web server, which you can access by pointing a web browser to your robot’s IP address.  From there, there is a logs tab, if you are able to access it, can you attach the log.

If your robot is on the network, is any other ROS software running on your network?  We have sometimes seen issues where different usages of the same topic (such as /Odom) can cause crashes in rclcpp or multiple robots on the same network conflict with each other and you have to use namespace/domain to isolate them.

**Comment by Affan60** [comment, marked as answer]
Hi,

Thank you for the response. The robot was able to boot successfully when unpacked. I connected the robot to my laptop through wifi connection and updated the firmware as mentioned in the setup guide.

Then I tried to connect it with ROS2 but I could not access the robot topics on ROS. So I changed the RMW on webpage of robot and pressed save and restart. Since then robot is not able to boot. So robot is not connected to any network.

Since robot is not connected to any network or my laptop, I am unable to connect it to the web server.

Regards,

Affan

**Reply by justinIRBT** [reply]
Do you have a raspberry pi or Nvidia Jetson or a computer with an OTG port that you could plug into the usb-c port of the robot to access the robot over the 192.168.186.2 address?  The create is a host USB device, so it needs to connect to something that can run as a client (as the OTG ports on the raspberry pi and Jetson allow).

To confirm, when you changed the RMW implement, you used the 2 button press and connected your laptop to the Create-#### AP to access the web browser and you changed the RMW to fastrtps from the cycloneDDS default?

**Reply by shamlian** [reply]
Edit: try Justin's suggestion before this one.

Switching middleware shouldn't cause your robot to never boot again, nor should it cause it to disconnect from your network if it had already been connected. Had you been connected to your local Wi-Fi network, or had you been doing all of this with the robot acting as an access point? One final thing to try would be to fully reset the system by removing the battery for at least fifteen minutes. Take the bottom cover off by removing the five stars marked with asterisks in [this drawing](https://iroboteducation.github.io/create3_docs/hw/mechanical/#bottom-view), and then you will be able to access the green battery. After fifteen minutes, replace the battery and bottom cover. The robot should appear as it did when you received it (fully off). At this point, place it on the docking station. The robot should boot normally. If it does not, please e-mail us at create@irobot.com so that we can arrange for the next steps.

**Reply by Affan60** [reply]
I changed the RWM to fastrtps from the cyclonedds and clicked the restart. My laptop disconnected from wifi connection to iRobot and connected my  wifi internet connection.  The went to reboot and not completed the booting stage since then.

I did remove the battery for around twenty (20) minutes and replace the battery again. when I placed the robot on charging dock, it went to booting mode but not completed the booting process. 



I have raspberry Pi. I connected the robot to raspberry Pi through OTG cable and moved the BLE/USB toggle button on adapter board to USB. I entered the 192.168.186.2  on chrome. It did not show anything. 

I think as robot is under booting stage, it will not communicate. It has to complete the booting stage to able to communicate by any mean. The robot is not able to complete its booting stage for last 24 hours. Its in the same situation, white ring keeps swirling with no sound.

**Reply by justinIRBT** [reply]
The application/spinning white light ring is separate from the OS and web server, so it is possible we can still get to the web server to recover the system even though the application is crashing.  Did you give your raspberry pi a 192.168.186.3 address as described in steps 2-5 here: https://iroboteducation.github.io/create3_docs/setup/pi4/
If you run ifconfig on your pi, do you see the usb0 interface with 192.168.186.3 address?  Can you ping 192.168.186.2 from the pi?

**Reply by shamlian** [reply]
I am going to try to reproduce this issue. Stand by.

**Reply by shamlian** [reply]
I factory reset my robot, put it in provisioning mode, changed the RMW, saved, and restarted application. The robot disconnected me from its access point, but then successfully booted, and I was able to subsequently put the robot into the provisioning mode again and connect it to my network and see its topics. Did you have any luck connecting over USB?

**Reply by Affan60** [reply]
I am able to connect robot with raspberry pi via USB-C cable and access 192.168.186.2 through links. It shows settings of my robot. When I try to make any changes, it does not respond. For example, if I click on reboot robot, it shows error "Oops, something went wrong".  How do I proceed further?

**Reply by shamlian** [reply]
Let's first try switching your robot back to CycloneDDS. From the Raspberry Pi, try
`curl -X POST -d "ros_domain_id=0&ros_namespace=&rmw_implementation=rmw_cyclonedds_cpp" "http://192.168.186.2/ros-config-save-main"`

**Reply by Affan60** [reply]
But can't access the 192.168.186.3

**Reply by Affan60** [reply]
> Let's first try switching your robot back to CycloneDDS. From the Raspberry Pi, try `curl -X POST -d "ros_domain_id=0&ros_namespace=&rmw_implementation=rmw_cyclonedds_cpp" "http://192.168.186.2/ros-config-save-main"`

ok. On it

**Reply by shamlian** [reply]
Another thing that would be helpful is a look at your logs, which you can download onto your Pi by typing `wget  http://192.168.186.2/logs-raw` ; if you post the logs-raw file here, we can maybe gain some insight into what is going on.

**Reply by Affan60** [reply]
Robot completed its boot up process. I was in process to switch it back to cyclonedds (not completed the process) and I heard the robot's happy sound. It happened without me entering curl -X POST -d "ros_domain_id=0&ros_namespace=&rmw_implementation=rmw_cyclonedds_cpp" "http://192.168.186.2/ros-config-save-main"

However, I clicked the "reboot the robot" and "Restart the application" multiple times when I was on 192.168.186.2.  

I checked the robot configurations. It shows that robot was on cyclonedds. It did not complete the change of RMW at the first place. 

All in all, robot is working now.

**Reply by shamlian** [reply]
I'm glad to hear the robot is working! Would you mind grabbing the logs for us before you reboot again, so that we can try to understand what might have happened?

**Reply by Affan60** [reply]
[messages.txt](https://github.com/iRobotEducation/create3_docs/files/8756820/messages.txt)

Logs are attached. Thanks for all the help

**Reply by shamlian** [reply]
Also -- while you've got the robot up -- I would recommend connecting it to your local Wi-Fi network, just so that debugging becomes easier if necessary in the future.

**Reply by shamlian** [reply]
One final thought -- it's not recommended to use this robot in access point mode for an extended period of time. If your intention is to connect over Wi-Fi, you should connect to an external access point.
